linera-base/src/identifiers.rs

```diff
@@ -183,18 +183,16 @@ pub enum BlobType {
 
 impl Display for BlobType {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        match serde_json::to_string(self) {
-            Ok(s) => write!(f, "{}", s),
-            Err(_) => Err(fmt::Error),
-        }
+        write!(f, "{:?}", self)
     }
 }
 
 impl FromStr for BlobType {
     type Err = anyhow::Error;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        serde_json::from_str(s).with_context(|| format!("Invalid BlobType: {}", s))
+        serde_json::from_str(&format!("\"{s}\""))
+            .with_context(|| format!("Invalid BlobType: {}", s))
```

Comment:
Doing this manually with `match` would avoid the string allocation.

Reply:
Well I didn't want to write all the case